piccolo/query/methods/insert.py

```diff
@@ -54,6 +54,14 @@ def on_conflict(
             t.List[t.Union[Column, t.Tuple[t.Union[str, Column], t.Any]]]
         ] = None,
     ) -> Self:
+        if (
+            self.engine_type == "sqlite"
+            and self.table._meta.db.get_version_sync() < 3.34
```

Comment:
Should be version less than 3.24 `and self.table._meta.db.get_version_sync() < 3.24`

Reply:
Oops - good catch!